### ISSUE TITLE
Multiple dex files define Lokhttp3/Address (gradle erro while signing .apk)

### ISSUE THREAD

**Issue by mecoFarid** [square/okhttp#3150]
So I have included new dependencies:

   ```
    compile 'com.google.code.gson:gson:2.8.0'
    compile 'com.squareup.retrofit2:converter-gson:2.1.0'
    compile 'com.squareup.retrofit2:retrofit:2.1.0'
```

after that I tried to generate signed .apk but it gave me error

> Error:Error converting bytecode to dex:
> Cause: com.android.dex.DexException: Multiple dex files define Lokhttp3/Address;
> Error:Execution failed for task ':app:transformClassesWithDexForProdRelease'.
> com.android.build.api.transform.TransformException: com.android.ide.common.process.ProcessException: java.util.concurrent.ExecutionException: java.lang.UnsupportedOperationException

I believe enabling multidex would solve problem, but I don't want to go with that as it slows down the app.

***Important Note: 
I have added okhtp-3.2.jar file to my app  long before I added these dependencies.

So, I would appreciate any solution other than multidex enabling method :)

**Comment by mecoFarid** [comment]
So, just found solution:
```
 compile 'com.squareup.retrofit2:converter-gson:2.1.0'
 compile 'com.squareup.retrofit2:retrofit:2.1.0'
```

already pulls following modules:
```
okhttp-3.2.0
okio-1.6.0
retrofit-2.0.1
converter-gson-2.0.1
```
I, just, simply removed `okhttp-3.2.0.jar` and `okio-1.6.0.jar` files

**Comment by naturalwarren** [comment]
Why does removing compile dependencies that are already included transitively (okio, okhttp) bring you below the dex limit?

**Comment by mecoFarid** [comment]
@naturalwarren as I said 

> I believe enabling multidex would solve problem
so apparently, that was not case. Digging a little deeper, I stated that:

> I have added okhttp-3.2.jar file to my app long before I added these dependencies.  
Adding below dependencies:
```
 compile 'com.squareup.retrofit2:retrofit:2.1.0'
```
adds `okhttp-3.2.0` module to app as external library (it's retrofits default configuration) which was already added from my `okhttp-3.2.jar` file. So, I was inadvertently adding `okhttp` to my app from `okhttp-3.2.jar` file and from `compile 'com.squareup.retrofit2:retrofit:2.1.0'`. That way, I was getting error stating that "Multiple dex files define Lokhttp3/Address => you have already defined this method in your app". I simply deleted my .jar file and proceeded with dependencies. I believe, I'm clear enough :)
